[docs] Improve bundle size option 2 advantage wording (#17577)
mui-org/material-ui: docs/src/pages/guides/minimizing-bundle-size/minimizing-bundle-size.md

### Option 1

You can use path imports to avoid pulling in unused modules.
For instance, use:

```js
// 🚀 Fast
import Button from '@material-ui/core/Button';
import TextField from '@material-ui/core/TextField';
```

instead of top level imports (without a Babel plugin):

```js

> // 🐌 Slow bootstrap

Reviewer: Not sure about the use of the word bootstrap in this context...

import { Button, TextField } from '@material-ui/core';
```

> This is the option we document in **all** the demos because it requires no configuration.

Reviewer: ```suggestion
This is the option we document in all the demos, since it requires no configuration.
```

### Option 2

This option provides the best User Experience and Developer Experience:

> - UX: the Babel plugin makes it as fast as the option 1.

Reviewer: ```suggestion
- UX: The Babel plugin makes it as fast as option 1.
```

If the performance (or rather the bundle size) is the same as option 1, how us the UX "the best" with this option?

Reviewer 2: Great point. I'm changing it to:
```diff
-This option provides the best User Experience and Developer Experience:
+This option provides the best User Experience and Developer Experience tradeoff:
```

Author: This change makes it again unclear.
If someone wants to have the fastest imports, then this change again confuses it to being labeled as just a 'tradeoff', and justifies promoting using direct imports (and supporting it with a codemode)

Reviewer 2: Oh, you made me realize that the wording is wrong. These are two DX pros, the UX pros are about forcing top-level tree shaking for bundler that doesn't support it. Maybe it's marginal and we should only focus on the DX tradeoff. The tradeoff is the need for a Babel configuration step to get fast startup time in exchange.

To answer your concern, isn't the list below making it clear?

> - DX: the syntax reduces the duplication of information. You don't have to duplicate the package name between two imports, you don't have duplicate the module name for each import.

Reviewer: ```suggestion
- DX: This syntax reduces the duplication of code, requiring only a single import for multiple modules.
```
